src/odemis/acq/move.py

```diff
@@ -218,84 +347,64 @@ def _doCryoSwitchSamplePosition(future, stage, focus, align, target):
         # Create axis->pos dict from target position given smaller number of axes
         filter_dict = lambda keys, d: {key: d[key] for key in keys}
 
-        # Initial submoves for all procedures is to park focus & aligner to safe position
-        if not _isNearPosition(focus.position.value, focus_deactive, focus.axes):
-            sub_moves.append((focus, focus_deactive))
-        if not _isNearPosition(align.position.value, align_deactive, align.axes):
-            run_sub_move(future, align, align_deactive)
-
         current_label = getCurrentPositionLabel(current_pos, stage)
         if target == LOADING:
             if current_label is UNKNOWN and stage_referenced:
                 logging.warning("Moving stage to loading while current position is unknown.")
-            if abs(stage_deactive['rx']) > ATOL_ROTATION_POS:
+            if abs(target_pos[LOADING]['rx']) > ATOL_ROTATION_POS:
                 raise ValueError(
                     "Absolute value of rx for FAV_POS_DEACTIVE is greater than {}".format(ATOL_ROTATION_POS))
 
             # Check if stage is not referenced:
-            # 1. reference focus if not already referenced
-            # 2. Move focus to deactive position
-            # 3. reference aligner if not already referenced
-            # 4. Move aligner to deactive position
-            # 5. reference stage
+            # park aligner then reference the stage
             if not stage_referenced:
-                if not all(focus.referenced.value.values()):
-                    run_reference(future, focus)
-                run_sub_move(future, focus, focus_deactive)
-                if not all(align.referenced.value.values()):
-                    run_reference(future, align)
-                run_sub_move(future, align, align_deactive)
+                cryoSwitchAlignPosition(LOADING).result()
```

Comment:
A small comment on what is happening here would be nice (put to loading position and wait until task is finished)?